internal/server/http/dump.go

```diff
@@ -0,0 +1,116 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"strings"
+)
+
+// RequestData struct to hold request data
+type RequestData struct {
+	Method string              `json:"method"`
+	Host   string              `json:"host"`
+	URL    string              `json:"url"`
+	Header map[string][]string `json:"header"`
+	Body   string              `json:"body"`
+}
+
+func GetRequestData(r *http.Request) *RequestData {
+	bodyBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Error reading request body: %v", err)
+		return nil
+	}
+	r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes)) // Reset the body
+
+	return &RequestData{
+		Method: r.Method,
+		Host:   r.Host,
+		URL:    r.URL.String(),
+		Header: r.Header,
+		Body:   string(bodyBytes),
```

Comment:
The body bytes can be binary, casting it directly into string without encoding (base64 etc) will be problematic in JSON.

Reply:
I am going to have to think on this one. The whole point of logging it out is to make it easily readable, if we base64 encode it then it makes it so it is no longer human readable without an extra step. I will see what I can come up with.

Reply:
Added base64 encoding of the body for the logging and the writing: https://github.com/friendsofgo/killgrave/pull/170/commits/acaa33c762ae837fa4b5a453fd400e85795b1429#diff-a20d872873fde6ebcd5539ac2f8fbcc90f715fcdb654b4fa499ccdd6dc542d22R179

Reply:
Yes, if the body is binary (based on Content-Type) header, should be converted to base64 ....

Reply:
I created a function in this commit to check for common content type headers that are binary and base64 encode them: https://github.com/friendsofgo/killgrave/pull/170/commits/ac2569aa80c9be529051471288302c639d0bc888

If I am missing any headers that should be considered binary please let me know and I will add them.